__openzfs: update auto key loading for OpenZFS v2.1.3

sideeffect42 opened the pull request:
Starting with OpenZFS 2.1.3 `MOUNT_EXTRA_OPTIONS='-l'` does not auto load ZFS keys anymore.
A new option was introduced which, however, does not work with older versions.

The explorer detects the ZFS version by calling `zfs version`.
However, because explorers always run first, `zfs(1)` won't be available on the first skonfig run.
Since the version switch is in the `manifest`, the common hack-around to query the version again in `code-remote` does not work.

What should be done in such cases?

I could think of two possibilities:
1. guess (assume latest? guess based on OS version?)
2. try to query the package manager based on the OS and hope that's the version that gets installed.

4nd3r commented:
If not installed then assume latest. And fix it on second pass, if older gets installed.

sideeffect42 commented:
> If not installed then assume latest. And fix it on second pass, if older gets installed.

I updated the code, albeit a bit differently.
Instead of assuming the latest version I added `if`s which skip version-dependent code if the version could not be determined.

I didn't want to add code to undo "incorrectly" applied changes.
